src/fluree/db/json_ld/transact.cljc

```diff
@@ -625,14 +625,31 @@
                                        (<? (query-range/index-range db-before :spot = [sid const/$rdf:type]
                                                                     {:flake-xf (map flake/o)}))))
                 class-shapes   (<? (shacl/class-shapes db-before classes))
+                ;; these target objects in s-flakes
+                pid->ref-flakes (->> s-flakes
```

Comment:
Is this work for `pid->ref-flakes` and `o-pred-shapes` also only needed when we have target-objects-of shapes? 

It kinda looks like there's a chunk of work that could just be its own fn that's only called when `has-target-objects-of-shapes` is true